### PR TITLE
clubhouse: Hook up sounds for entry button

### PR DESCRIPTION
**Pull request by cfoch**
https://phabricator.endlessm.com/T24716

### CODE REVIEW

**Comment by cosimoc** [comment]
@joaquimrocha implemented the same concept of the sound-able button in https://github.com/endlessm/gnome-shell/pull/394. We should indeed settle on only one version of the two.

**Comment by joaquimrocha** [comment]
My approach is slightly different: it's an independent class for playing one-off sounds (i.e. no overlapping) which uses the logic that was already in the CodeView. I think this way it can be used for other events too, not only on buttons.

**Comment by cfoch** [comment]
Should I still use the Soundable button on top of the SoundItem class?

**Comment by cosimoc** [comment]
@cfoch Yes sounds good. Please rework your PR to use the SoundItem API that Joaquim introduced.

File: js/ui/soundable.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            GObject.ParamFlags.READWRITE,
+            ''
+        )
+    },
+
+    _init: function(params) {
+        this.parent(params);
+
+        if (this.stop_hover_sound_on_click === undefined)
+            this.stop_hover_sound_on_click = false;
```
Comment (cosimoc) [inline js/ui/soundable.js]: If we make the `stop-hover-on-sound-click` property a construct one (by setting `GObject.ParamFlag.CONSTRUCT`), you should not need this, as the setter will be called during initialization.

File: js/ui/soundable.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            '',
+            GObject.ParamFlags.READWRITE,
+            ''
+        ),
+        'stop-hover-sound-on-click': GObject.ParamSpec.boolean(
+            'stop-hover-sound-on-click',
+            '',
+            '',
+            GObject.ParamFlags.READWRITE,
+            ''
```
Comment (cosimoc) [inline js/ui/soundable.js]: Should be `false` since this is a boolean property

File: js/ui/soundable.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        this._hover_sound_event_id = value;
+        this._hover_sound_item =
+            new SoundServer.SoundItem(this._hover_sound_event_id);
+        this.connect('notify::hover', this._onHoverChanged.bind(this));
+    },
+
+    set click_sound_event_id(value) {
+        this._click_sound_event_id = value;
+        this.connect('clicked', () => {
+            if (this.stop_hover_sound_on_click) {
```
Comment (cosimoc) [inline js/ui/soundable.js]: Remove curly braces for single-line blocks

File: js/ui/soundable.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                this._stopHoverSound();
+            }
+            SoundServer.getDefault().play(this._click_sound_event_id);
+        });
+    },
+
+    _onHoverChanged: function() {
+        if (this.hover) {
```
Comment (cosimoc) [inline js/ui/soundable.js]: Remove curly braces here too

File: js/ui/components/clubhouse.js [hunk 5]
```diff
@@ -41,6 +41,7 @@ const NotificationDaemon = imports.ui.notificationDaemon;
 const SideComponent = imports.ui.sideComponent;
 const SoundServer = imports.misc.soundServer;
 const Tweener = imports.ui.tweener;
+const Soundable = imports.ui.soundable;
```
Comment (cosimoc) [inline js/ui/components/clubhouse.js]: Please keep alphabetical order

File: js/js-resources.gresource.xml [hunk 6]
```diff
@@ -72,6 +72,7 @@
     <file>ui/messageList.js</file>
     <file>ui/modalDialog.js</file>
     <file>ui/mpris.js</file>
+    <file>ui/soundable.js</file>
```
Comment (cosimoc) [inline js/js-resources.gresource.xml]: Put this in the endless-specific section (also alphabetical order)

File: js/ui/soundable.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+            '',
+            '',
+            GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT,
+            false
+        )
+    },
+
+    _init: function(params) {
+        this.parent(params);
+    },
```
Comment (cosimoc) [inline js/ui/soundable.js]: This method does nothing more than the default implementation now, and can be removed.

**Pull request merged**